src/bp-messages/bp-messages-filters.php

```diff
@@ -727,6 +730,77 @@ function bp_recipients_recipient_get_where_conditions_callback( $where_condition
 	return $where_conditions;
 }
 
+/**
+ * Function will check content empty or not for the media, document and gif.
+ * If content will empty then return true and allow empty content in DB for the media, document and gif.
+ *
+ * @param bool         $validated_content Boolean from filter.
+ * @param string       $content           Message content.
+ * @param array|object $post              Request object.
+ *
+ * @return bool
+ */
+function bb_check_is_message_content_empty( $validated_content, $content, $post ) {
+	if ( ( empty( $post['content'] ) || empty( $post['message_content'] ) ) && ( isset( $post['gif_data'] ) || isset( $post['media'] ) || isset( $post['document'] ) || isset( $post['video'] ) ) ) {
+		return true;
+	} elseif ( ! empty( $post['content'] ) || ! empty( $post['message_content'] ) ) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+/**
+ * Function will check content empty or not for the media, document and gif.
+ * If content will empty then return true and allow empty content in DB for the media, document and gif.
+ *
+ * @param bool         $validated_content Boolean from filter.
+ * @param string       $content           Message content.
+ * @param array|object $id                Message ID.
+ *
+ * @return bool
+ */
+function bb_check_is_get_message_content_empty( $validated_content, $content = '', $message_id ) {
+
+	if ( ! empty( $content ) ) {
+		return true;
+	}
+
+	if ( bp_is_active( 'media' ) && bp_is_messages_media_support_enabled() ) {
```

Comment:
@milanhirapra Instead of adding media-related code here, this should be added in the media file. 
You can apply filters multiple times with different piority.

Reply:
@dipskakadiya I added filters to related component files instead of message filter file. Please check this video: https://vimeo.com/600805119/fdd080abeb